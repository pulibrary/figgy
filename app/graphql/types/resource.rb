# frozen_string_literal: true
module Types::Resource
  include Types::BaseInterface
  description "A resource in the system."
  orphan_types Types::ScannedResourceType, Types::FileSetType, Types::ScannedMapType

  field :id, String, null: true
  field :label, String, null: true
  field :viewing_hint, String, null: true
  field :url, String, null: true
  field :members, [Types::Resource], null: true
  field :source_metadata_identifier, String, null: true
  field :thumbnail, Types::Thumbnail, null: true

  definition_methods do
    def resolve_type(object, _context)
      "Types::#{object.class}Type".constantize
    end
  end

  def members
    @members ||= Wayfinder.for(object).members
  end

  def manifest_url
    helper.polymorphic_url([:manifest, object])
  end

  def url
    @url ||= helper.show_url(object)
  end

  # We need to centralize logic for navigating a MVW's members to find a
  # thumbnail file set. This is a hack to use the helper's logic for doing that.
  # refactor ticketed as https://github.com/pulibrary/figgy/issues/1708
  def helper
    @helper ||= ManifestBuilder::ManifestHelper.new.tap do |helper|
      helper.singleton_class.include(ThumbnailHelper)
      helper.define_singleton_method(:image_tag) do |url, _opts|
        url
      end
      helper.define_singleton_method(:image_path) do |url|
        url
      end
    end
  end

  def thumbnail
    return if object.try(:thumbnail_id).blank? || thumbnail_resource.blank?
    {
      id: thumbnail_resource.id.to_s,
      thumbnail_url: helper.figgy_thumbnail_path(thumbnail_resource),
      iiif_service_url: helper.figgy_thumbnail_path(thumbnail_resource).gsub("/full/!200,150/0/default.jpg", "")
    }
  end

  def thumbnail_resource
    @thumbnail_resource ||= query_service.find_by(id: object.try(:thumbnail_id).first)
  rescue Valkyrie::Persistence::ObjectNotFoundError
    nil
  end

  def query_service
    Valkyrie::MetadataAdapter.find(:indexing_persister).query_service
  end
end
