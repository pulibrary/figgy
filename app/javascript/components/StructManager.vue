<template>
  <div class="lux-structManager">
    <toolbar @cards-resized="resizeCards($event)" />
    <div
      class="lux-sidePanel"
    ></div>
    <div
      class="lux-galleryPanel"
      >
      <gallery
        class="lux-galleryWrapper"
        :card-pixel-width="cardPixelWidth"
        :gallery-items="galleryItems"
      />
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex'
import Toolbar from './StructManagerToolbar'

/**
 * OrderManager is a tool for reordering thumbnails that represent members of a complex object (a book, CD, multi-volume work, etc.).
 * Complex patterns like OrderManager come with their own Vuex store that it needs to manage state.
 * The easiest way to use the OrderManager is to simply pass a resource in as a prop.
 * You can see how this is done in the live code example at the end of this section.
 *
 * However you will still need to load the corresponding
 * Vuex module, *resourceModule*. Please see [the state management documentation](/#!/State%20Management) for how to manage state in complex patterns.
 */
export default {
  name: 'StructManager',
  status: 'ready',
  release: '1.0.0',
  type: 'Pattern',
  components: {
    'toolbar': Toolbar,
  },
  props: {
    /**
     * The resource object in json format.
     */
    resourceObject: {
      type: Object,
      default: null
    },
    /**
     * The resource id. Requires host app to have async lookup of resource.
     */
    resourceId: {
      type: String,
      default: null
    },
    defaultThumbnail: {
      type: String,
      default: 'https://picsum.photos/600/300/?random'
    }
  },
  data: function () {
    return {
      cardPixelWidth: 300,
      captionPixelPadding: 9
    }
  },
  computed: {
    galleryItems () {
      return this.resource.members.map(member => ({
        id: member.id,
        caption: member.label,
        service:
          member['thumbnail'] && typeof (member.thumbnail.iiifServiceUrl) !== 'undefined'
            ? member.thumbnail.iiifServiceUrl
            : this.defaultThumbnail,
        mediaUrl:
          member['thumbnail'] && typeof (member.thumbnail.iiifServiceUrl) !== 'undefined'
            ? member.thumbnail.iiifServiceUrl + '/full/300,/0/default.jpg'
            : this.defaultThumbnail,
        viewingHint: member.viewingHint
      }))
    },
    selectedTotal () {
      return this.gallery.selected.length
    },
    isMultiVolume () {
      return this.$store.getters.isMultiVolume
    },
    ...mapState({
      resource: state => state.ordermanager.resource,
      gallery: state => state.gallery
    }),
    loading: function () {
      return this.resource.loadState !== 'LOADED'
    },
    saved () {
      return this.resource.saveState === 'SAVED'
    },
    saveError () {
      return this.resource.saveState === 'ERROR'
    },
    isLoading () {
      return this.resource.saveState === 'SAVING'
    }
  },
  beforeMount: function () {
    if (this.resourceObject) {
      // if props are passed in set the resource on mount
      this.$store.commit('SET_RESOURCE', this.resourceObject)
    } else {
      let resource = { id: this.resourceId }
      this.$store.commit('CHANGE_RESOURCE_LOAD_STATE', 'LOADING')
      this.$store.dispatch('loadImageCollectionGql', resource)
    }
  },
  methods: {
    resizeCards: function (event) {
      this.cardPixelWidth = event.target.value
      if (this.cardPixelWidth < 75) {
        this.captionPixelPadding = 0
      } else {
        this.captionPixelPadding = 9
      }
    }
  }
}
</script>
<style lang="scss">
.lux-resourceTitle {
  background-color: grey;
  margin-bottom: 10em;
}

.lux-title {
  font-weight: bold;
}
.lux-structManager {
  position: relative;
  height: 80vh;
}
.lux-structManager .lux-heading {
  margin: 12px 0 12px 0;
  line-height: 0.75;
  color: #001123;
}
.lux-structManager h2 {
  letter-spacing: 0;
  font-size: 24px;
}
.lux-sidePanel {
  position: absolute;
  top: 70px;
  left: 0px;
  height: 85%;
  width: 28.5%;
  border: 1px solid #ddd;
  border-radius: 4px;

  padding: 0 30px 0 30px;
  // height: 100%;
  overflow-y: scroll;
}
.lux-sidePanel .lux-input {
  display: block;
}
.lux-galleryPanel {
  position: absolute;
  top: 70px;
  left: 30%;
  height: 85%;
  width: 70%;
  border-radius: 4px;
  border: 1px solid #ddd;
}
.lux-icon {
  margin: auto;
}
.lux-galleryWrapper {
  overflow: auto;
  height: calc(100% - 80px);
  border-radius: 4px;
  margin-bottom: 80px;
  clear: both;
}

.loader {
  position: absolute;
  width: 100%;
  height: 100%;
  text-align: center;
  padding-bottom: 64px;
  z-index: 500;
  margin-top: -16px;
}
.loader .galleryLoader {
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.85);
  display: flex;
}
.loader .galleryLoader .lux-loader {
  margin: auto;
}
</style>
