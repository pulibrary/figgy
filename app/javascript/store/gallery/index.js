export const galleryState = {
  items: [],
  selected: [],
  cut: [],
  changeList: [],
  ogItems: [],
}

export const galleryMutations = {
  CUT(state, itemArray) {
    state.cut = [...itemArray]
  },
  PASTE(state, itemArray) {
    state.items = [...itemArray]
  },
  SELECT(state, itemArray) {
    state.selected = [...itemArray]
  },
  SET_GALLERY(state, items) {
    state.items = items
    state.ogItems = items
  },
  SORT_ITEMS(state, value) {
    state.items = [...value]
  },
  UPDATE_CHANGES(state, changeList) {
    state.changeList = [...changeList]
  },
  UPDATE_ITEMS(state, items) {
    state.items = [...items]
  },
}

export const galleryModule = {
  state: galleryState,
  mutations: galleryMutations,
}
