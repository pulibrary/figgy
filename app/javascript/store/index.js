import Vue from 'vue/dist/vue.esm'
import Vuex from 'vuex'
import actions from './vuex/actions'
import {modules} from 'lux-design-system'
Vue.use(Vuex)

const store = new Vuex.Store({
  actions,
  modules: {
    ordermanager: modules.resourceModule,
    gallery: modules.galleryModule,
  }
})

export default store
