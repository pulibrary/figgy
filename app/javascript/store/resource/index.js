export const resourceState = {
  resource: {
    id: "",
    resourceClassName: "",
    bibId: "",
    label: "Resource not available.",
    thumbnail: "",
    startCanvas: "",
    isMultiVolume: false,
    viewingHint: null,
    viewingDirection: null,
    members: [],
    loadState: "NOT_LOADED",
    saveState: "NOT_SAVED",
    ogState: {},
  },
}

export const resourceMutations = {
  APPLY_STATE(state) {
    state.gallery.ogItems = state.gallery.items
    state.gallery.changeList = []
    state.resource.ogState = {
      startCanvas: resource.startPage,
      thumbnail: resource.thumbnail,
      viewingHint: resource.viewingHint,
      viewingDirection: resource.viewingDirection,
    }
    state.resource.saveState = "NOT_SAVED"
  },
  CHANGE_RESOURCE_LOAD_STATE(state, loadState) {
    state.resource.loadState = loadState
  },
  SAVED_STATE(state, saveStatus) {
    state.resource.saveState = saveStatus
  },
  SET_RESOURCE(state, resource) {
    state.resource.id = resource.id
    state.resource.resourceClassName = resource.__typename
    state.resource.label = resource.label
    state.resource.startCanvas = resource.startPage
    state.resource.viewingHint = resource.viewingHint
    state.resource.viewingDirection = resource.viewingDirection
    state.resource.thumbnail = resource.thumbnail != null ? resource.thumbnail.id : null
    state.resource.members = resource.members

    const items = resource.members.map(member => ({
      id: member.id,
      viewingHint: member.viewingHint != null ? member.viewingHint : "single",
      caption: member.label, // member.__typename + " : " + member.id,
      service:
        member["thumbnail"] && typeof(member.thumbnail.iiifServiceUrl) != "undefined"
          ? member.thumbnail.iiifServiceUrl
          : Global.figgy.resource.defaultThumbnail,
      mediaUrl:
        member["thumbnail"] && typeof(member.thumbnail.iiifServiceUrl) != "undefined"
          ? member.thumbnail.iiifServiceUrl + "/full/300,/0/default.jpg"
          : Global.figgy.resource.defaultThumbnail,
    }))
    state.gallery.items = items
    state.gallery.ogItems = items
    state.resource.loadState = "LOADED"
    state.resource.ogState = {
      startCanvas: resource.startPage,
      thumbnail: resource.thumbnail != null ? resource.thumbnail.id : null,
      viewingHint: resource.viewingHint,
      viewingDirection: resource.viewingDirection,
    }
  },
  UPDATE_STARTCANVAS(state, startCanvas) {
    state.resource.startCanvas = startCanvas
  },
  UPDATE_THUMBNAIL(state, thumbnail) {
    state.resource.thumbnail = thumbnail
  },
  UPDATE_VIEWDIR(state, viewDir) {
    state.resource.viewingDirection = viewDir
  },
  UPDATE_VIEWHINT(state, viewHint) {
    state.resource.viewingHint = viewHint
  },
}

export const resourceGetters = {
  getMemberCount: state => {
    return state.resource.members.length
  },
  isMultiVolume: state => {
    const volumes = state.resource.members.filter(member => member.__typename === "ScannedResource")
    return volumes.length > 0 ? true : false
  },
  orderChanged: state => {
    let ogOrder = JSON.stringify(state.gallery.ogItems.map(item => item.id))
    let imgOrder = JSON.stringify(state.gallery.items.map(item => item.id))
    return ogOrder !== imgOrder
  },
  stateChanged: (state, getters) => {
    var propsChanged = []
    propsChanged.push(state.resource.ogState.thumbnail !== state.resource.thumbnail)
    propsChanged.push(state.resource.ogState.startCanvas !== state.resource.startCanvas)
    propsChanged.push(state.resource.ogState.viewingHint !== state.resource.viewingHint)
    propsChanged.push(state.resource.ogState.viewingDirection !== state.resource.viewingDirection)
    propsChanged.push(state.gallery.changeList.length > 0)
    propsChanged.push(getters.orderChanged)
    if (propsChanged.indexOf(true) > -1) {
      return true
    } else {
      return false
    }
  },
}
