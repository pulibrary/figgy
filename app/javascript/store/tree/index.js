export const treeState = {
  selected: null,
}

export const treeMutations = {
  SELECT(state, selectedId) {
    state.selected = selectedId
  },
}
