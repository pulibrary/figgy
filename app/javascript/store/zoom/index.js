export const zoomState = {
  zoomed: null,
}

export const zoomMutations = {
  ZOOM(state, obj) {
    state.zoomed = obj
  },
  RESET_ZOOM(state) {
    state.zoomed = null
  },
}

export const zoomGetters = {
  getIsZoomed: state => {
    return state.zoomed
  },
}
