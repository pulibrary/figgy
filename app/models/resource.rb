# frozen_string_literal: true
class Resource < Valkyrie::Resource
  def self.human_readable_type
    default = @_human_readable_type || name.demodulize.titleize
    I18n.translate("models.#{new.model_name.i18n_key}", default: default)
  end

  def self.can_have_manifests?
    false
  end

  def human_readable_type
    self.class.human_readable_type
  end

  def self.model_name
    ::ActiveModel::Name.new(self)
  end

  # Determines if this is an image resource
  # @return [TrueClass, FalseClass]
  def image_resource?
    false
  end

  # Determines if this is a geospatial resource
  # @return [TrueClass, FalseClass]
  def geo_resource?
    false
  end

  # Determines if this is a media resource
  # @return [TrueClass, FalseClass]
  def media_resource?
    false
  end

  # Virtual property used for stashing pre-loaded objects. Populated by a query.
  attr_accessor :loaded
  def loaded
    @loaded ||= {}
  end
end
