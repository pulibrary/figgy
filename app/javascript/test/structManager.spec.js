import Vuex from "vuex"
import { createLocalVue, mount, shallowMount } from "@vue/test-utils"
import StructManager from "../components/StructManager.vue"
import { resourceMutations, resourceGetters } from "../store/resource"
import { treeMutations } from "../store/tree"
import { zoomMutations, zoomGetters } from "../store/zoom"
import { modules } from 'lux-design-system'

// create an extended `Vue` constructor
const localVue = createLocalVue()
localVue.use(Vuex)

// Work-around in order to ensure that the Global object is accessible from the component
const Global = {
  figgy: {
    resource: {
      defaultThumbnail: "https://institution.edu/repository/assets/random.png"
    }
  }
}
window.Global = Global

let wrapper
let items = [
  {
    "id": "1",
    "caption": "a",
    "service": "a1_service",
    "mediaUrl": "a1_url",
    "viewingHint": "single"
  },
  {
    "id": "2",
    "caption": "b",
    "service": "b2_service",
    "mediaUrl": "b2_url",
    "viewingHint": null
  },
  {
    "id": "3",
    "caption": "c",
    "service": "c3_service",
    "mediaUrl": "c3_url",
    "viewingHint": null
  }
]

let resourceObject = {
  id: "aea40813-e0ed-4307-aae9-aec53b26bdda",
  label: "Resource with 3 files",
  viewingHint: "individuals",
  viewingDirection: "LEFTTORIGHT",
  startPage: "1",
  thumbnail: {
    id: "1",
    thumbnailUrl:
      "a1_url",
    iiifServiceUrl: "a1_service",
  },
  __typename: "ScannedResource",
  members: [
    {
      id: "1",
      label: "a",
      viewingHint: "single",
      thumbnail: {
        iiifServiceUrl:
          "a1_service",
      },
      __typename: "FileSet",
    },
    {
      id: "2",
      label: "b",
      viewingHint: null,
      thumbnail: {
        iiifServiceUrl:
          "b2_service",
      },
      __typename: "FileSet",
    },
    {
      id: "3",
      label: "c",
      viewingHint: null,
      thumbnail: {
        iiifServiceUrl:
          "c3_service",
      },
      __typename: "FileSet",
    },
  ],
}
//////////////////////////
let figgy_structure = {
  "id": null,
  "internal_resource": "Structure",
  "created_at": null,
  "updated_at": null,
  "new_record": true,
  "label": [
    "Table of Contents"
  ],
  "nodes": [
    {
      "id": null,
      "internal_resource": "StructureNode",
      "created_at": null,
      "updated_at": null,
      "new_record": true,
      "label": [
        "Chapter 1"
      ],
      "proxy": [],
      "nodes": [
        {
          "id": null,
          "internal_resource": "StructureNode",
          "created_at": null,
          "updated_at": null,
          "new_record": true,
          "label": ['c'],
          "proxy": [
            {
              "id": "3"
            }
          ],
          "nodes": []
        }
      ]
    }
  ]
}
//////////////////////////
let tree_structure = {
  "id": "aea40813-e0ed-4307-aae9-aec53b26bdda",
  "folders": [
    {
      "id": "1234567",
      "label": "Chapter 1",
      "file": false,
      "folders": [
        {
          "id": "3",
          "label": "c",
          "file": true,
          "folders": [],
          "service": "c3_service",
          "mediaUrl": "c3_service/full/300,/0/default.jpg",
          "viewingHint": null
        }
      ]
    }
  ],
  "label": "Table of Contents"
}
//////////////////////////
let unstructured_items = [
  {
    "id": "1",
    "caption": "a",
    "service": "a1_service",
    "mediaUrl": "a1_service/full/300,/0/default.jpg",
    "viewingHint": "single"
  },
  {
    "id": "2",
    "caption": "b",
    "service": "b2_service",
    "mediaUrl": "b2_service/full/300,/0/default.jpg",
    "viewingHint": null
  },
]
const tree = {
  state: {
    selected: null,
    cut: null,
    structure: { label: "Table of Contents", id: "123", folders: [] },
    modified: false,
    loadState: "NOT_LOADED",
    saveState: "NOT_SAVED",
  },
  mutations: treeMutations,
}

const zoom = {
  state: {
    zoomed: null,
  },
  mutations: zoomMutations,
  getters: zoomGetters,
}

const gallery = {
  state: {
    items: items,
    selected: [items[0]],
    cut: [],
    changeList: ["2"],
    ogItems: items,
  },
  mutations: modules.galleryModule.mutations,
}

const actions = {
  loadImageCollectionGql: vi.fn()
}

let resource = {
  state: {
    resource: {
      id: "",
      resourceClassName: "",
      bibId: "",
      label: "Resource not available.",
      thumbnail: "",
      startCanvas: "",
      isMultiVolume: false,
      viewingHint: null,
      viewingDirection: null,
      members: [],
      loadState: "NOT_LOADED",
      saveState: "NOT_SAVED",
      ogState: {},
    },
  },
  mutations: resourceMutations,
  getters: resourceGetters,
  actions: actions,
  modules: {
    gallery: gallery,
  },
}

let store = new Vuex.Store({
  modules: {
    ordermanager: resource,
    gallery: gallery,
    zoom: zoom,
    tree: tree,
  },
})

describe("StructManager.vue", () => {
  beforeEach(() => {

    wrapper = mount(StructManager, {
      localVue,
      store,
      propsData: {
        resourceObject: resourceObject,
        structure: figgy_structure,
      },
      stubs: [
        "toolbar",
        "struct-gallery",
        "deep-zoom",
        "tree",
        "wrapper",
        "alert",
        "controls",
        "loader",
      ],
    })
  })

  it("has the right gallery items", () => {
    const items = wrapper.vm.galleryItems
    expect(items[0].caption).toBe("a")
    expect(items[0].mediaUrl).toBe(
      "a1_service/full/300,/0/default.jpg"
    )
    expect(items[1].mediaUrl).toBe("b2_service/full/300,/0/default.jpg")
  })

  it("has the right selectedTotal", () => {
    expect(wrapper.vm.selectedTotal).toBe(1)
  })

  it("returns whether or not the resource is Loading", () => {
    expect(wrapper.vm.loading).toBe(false)
  })

  it("returns whether or not the structure is saved", () => {
    expect(wrapper.vm.saved).toBe(false)
  })

  it("returns whether or not the structure has a saveError", () => {
    expect(wrapper.vm.saveError).toBe(false)
  })

  it("resizes cards properly", () => {
    expect(wrapper.vm.cardPixelWidth).toBe(300)
    expect(wrapper.vm.captionPixelPadding).toBe(9)
    wrapper.vm.resizeCards({ target: { value: 70 } })
    expect(wrapper.vm.cardPixelWidth).toBe(70)
    expect(wrapper.vm.captionPixelPadding).toBe(0)
    wrapper.vm.resizeCards({ target: { value: 100 } })
    expect(wrapper.vm.captionPixelPadding).toBe(9)
  })

  it("calls the right action when no resourceObject is passed in", () => {
    // need to remount since the action only fires before mounting
    const wrapper2 = mount(StructManager, {
      localVue,
      store,
      propsData: {
        resourceId: "foo",
      },
      stubs: [
        "toolbar",
        "struct-gallery",
        "deep-zoom",
        "tree",
        "wrapper",
        "alert",
        "controls",
        "loader",
      ],
    })
    expect(actions.loadImageCollectionGql).toHaveBeenCalled()
  })

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it("renders the links thumbnail", () => {
    expect(wrapper.vm.galleryItems.length).toEqual(3)
    expect(wrapper.vm.galleryItems[0]['mediaUrl']).toEqual('a1_service/full/300,/0/default.jpg')
  })

  it("generates a 7-digit random id for tree nodes", () => {
    const id = wrapper.vm.generateId()
    expect(id.length).toEqual(7)
  })

  it("updates the state of the tree and gallery after resource data is loaded", () => {
    const generateIdStub = vi.fn(() => '1234567')
    wrapper.setMethods({ generateId: generateIdStub })
    // the loading of data should trigger the watcher which calls wrapper.vm.filterGallery(true),
    // but we can test the function
    wrapper.vm.filterGallery(true)
    expect(wrapper.vm.ga).toEqual(unstructured_items)
    console.log(wrapper.vm.resourceId)
    expect(wrapper.vm.s).toEqual(tree_structure)
  })

  it("displays an alert with a success message when the structure is SAVED", () => {
    const savedStub = vi.fn(() => true)
    wrapper.setMethods({ saved: savedStub })
    wrapper.vm.saved
    expect(wrapper.text()).toContain('Your work has been saved!')
  })

  it("does not display an alert with an error message when the state is NOT SAVED", () => {
    const saveErrorStub = vi.fn(() => true)
    wrapper.setMethods({ saveError: saveErrorStub })
    wrapper.vm.saveError
    expect(wrapper.text()).not.toContain('Sorry, there was a problem saving your work!')
  })

  //
  // it("deselects tree items when gallery item is clicked", () => {
  //   expect(wrapper.vm.galleryClicked).toBe(null)
  // })
  //
  // describe('when the resources do not have IIIF service URLs', () => {
  //
  //   let resourceObject = {
  //     id: "example2-id",
  //     label: "Resource with 1 files",
  //     viewingHint: "individuals",
  //     viewingDirection: "LEFTTORIGHT",
  //     startPage: "8ffd7a03-ec0e-46c1-a347-e4b19cb7839f",
  //     thumbnail: null,
  //     __typename: "ScannedResource",
  //     members: [
  //       {
  //     	  id: "8ffd7a03-ec0e-46c1-a347-e4b19cb7839f",
  //     	  label: "a",
  //     	  viewingHint: null,
  //     	  thumbnail: null,
  //     	  __typename: "FileSet",
  //     	}
  //     ],
  //     loadState: "LOADED",
  //     saveState: "NOT_SAVED",
  //     ogState: {}
  //   }
  //
  //   resource = {
  //     state: {
  //       resource: resourceObject
  //     },
  //     mutations: resourceMutations,
  //     getters: resourceGetters,
  //     actions: actions,
  //     modules: {
  //       gallery: gallery,
  //     }
  //   }
  //
  //   let store = new Vuex.Store({
  //     modules: {
  //       ordermanager: resource,
	//       gallery: gallery,
  //       zoom: zoom,
  //       tree: tree,
  //     }
  //   })
  //
  //   let wrapper = mount(StructManager, {
  //     localVue,
  //     store: store,
  //     propsData: {
  //       resourceObject: resourceObject,
	//       defaultThumbnail: Global.figgy.resource.defaultThumbnail
  //     },
  //     stubs: [
  //       "toolbar",
  //       "struct-gallery",
  //       "deep-zoom",
  //       "tree",
  //       "wrapper",
  //       "alert",
  //       "controls",
  //       "loader",
  //     ],
  //   })
  //
  //   it("renders the default Figgy thumbnail", () => {
  //     expect(wrapper.vm.defaultThumbnail).toEqual(Global.figgy.resource.defaultThumbnail)
  //     expect(wrapper.vm.galleryItems.length).toEqual(1)
  //     expect(wrapper.vm.galleryItems[0]['mediaUrl']).toEqual("https://institution.edu/repository/assets/random.png")
  //   })
  //
  //   it("does not display an alert with an error message when the state is NOT SAVED", () => {
  //     expect(wrapper.text()).not.toContain('Sorry, there was a problem saving your work!')
  //   })
  // })
  //
  // describe('when the resources errors on Save', () => {
  //
  //   let resourceObject = {
  //     id: "example3-id",
  //     label: "Resource with 1 files",
  //     viewingHint: "individuals",
  //     viewingDirection: "LEFTTORIGHT",
  //     startPage: "8ffd7a03-ec0e-46c1-a347-e4b19cb7839f",
  //     thumbnail: null,
  //     __typename: "ScannedResource",
  //     members: [
  //       {
  //     	  id: "8ffd7a03-ec0e-46c1-a347-e4b19cb7839f",
  //     	  label: "a",
  //     	  viewingHint: null,
  //     	  thumbnail: null,
  //     	  __typename: "FileSet",
  //     	}
  //     ],
  //     loadState: "LOADED",
  //     saveState: "ERROR",
  //     ogState: {}
  //   }
  //
  //   resource = {
  //     state: {
  //       resource: resourceObject
  //     },
  //     mutations: resourceMutations,
  //     getters: resourceGetters,
  //     actions: actions,
  //     modules: {
  //       gallery: gallery,
  //     }
  //   }
  //
  //   let store = new Vuex.Store({
  //     modules: {
  //       ordermanager: resource,
	//       gallery: gallery,
  //     }
  //   })
  //
  //   let wrapper = mount(StructManager, {
  //     localVue,
  //     store: store,
  //     propsData: {
  //       resourceObject: resourceObject,
	//       defaultThumbnail: Global.figgy.resource.defaultThumbnail
  //     },
  //     stubs: [
  //       "toolbar",
  //       "struct-gallery",
  //       "deep-zoom",
  //       "tree",
  //       "wrapper",
  //       "alert",
  //       "controls",
  //       "loader",
  //     ],
  //   })
  //
  //   it("displays an alert with an error message", () => {
  //     expect(wrapper.text()).toContain('Sorry, there was a problem saving your work!')
  //   })
  // })
})
