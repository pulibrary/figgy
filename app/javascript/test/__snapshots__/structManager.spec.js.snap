// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StructManager.vue > has the expected html structure 1`] = `
<div
  class="lux-structManager"
  data-v-5aca45d1=""
>
  <transition-stub
    data-v-5aca45d1=""
    name="fade"
  >
    <!---->
  </transition-stub>
  <!---->
  <!---->
  <toolbar-stub
    data-v-5aca45d1=""
    type="div"
  />
  <deep-zoom-stub
    class="deep-zoom"
    data-v-5aca45d1=""
    resourceid=""
    type="div"
    viewerid="viewer"
  />
  <div
    class="lux-sidePanel"
    data-v-5aca45d1=""
  >
    <div
      class="panelHeader"
      data-v-5aca45d1=""
    >
      <h2
        class="lux-heading h6"
        data-v-5aca45d1=""
      >
        Logical Structure
      </h2>
    </div>
    <tree-stub
      data-v-5aca45d1=""
      id=""
      jsondata="[object Object]"
    />
  </div>
  <div
    class="lux-galleryPanel"
    data-v-5aca45d1=""
  >
    <div
      class="panelHeader"
      data-v-5aca45d1=""
    >
      <h2
        class="lux-heading h6"
        data-v-5aca45d1=""
      >
        Unstructured Files
      </h2>
    </div>
    <struct-gallery-stub
      cardpixelwidth="300"
      class="lux-galleryWrapper"
      data-v-5aca45d1=""
      galleryitems="[object Object],[object Object],[object Object]"
    />
  </div>
</div>
`;
